lib/src/view/coordinate_training/coordinate_training_screen.dart

```diff
@@ -0,0 +1,533 @@
+                                  : LichessColors.good,
+                            ),
+                            FatButton(
+                              semanticsLabel: 'Abort Training',
+                              onPressed: ref
+                                  .read(
+                                    coordinateTrainingControllerProvider
+                                        .notifier,
+                                  )
+                                  .stopTraining,
+                              child: const Text(
+                                'Abort Training',
+                                style: Styles.bold,
+                              ),
+                            ),
+                          ],
+                        ),
+                      )
+                    else
+                      _Settings(
+                        onSideChoiceSelected: _setOrientation,
+                      ),
+                  ],
+                );
+              },
+            ),
+          ),
+          BottomBar(
+            children: [
+              BottomBarButton(
+                label: context.l10n.menu,
+                onTap: () => showAdaptiveBottomSheet<void>(
+                  context: context,
+                  builder: (BuildContext context) =>
+                      const _CoordinateTrainingMenu(),
+                ),
+                icon: Icons.tune,
+              ),
+              BottomBarButton(
+                icon: Icons.info_outline,
+                label: context.l10n.aboutX('Coordinate Training'),
+                onTap: () => _coordinateTrainingInfoDialogBuilder(context),
+              ),
+            ],
+          ),
+        ],
+      ),
+    );
+  }
+
+  void _onGuess(Square square) {
+    ref
+        .read(coordinateTrainingControllerProvider.notifier)
+        .guessCoordinate(square);
+
+    setState(() {
+      highlightLastGuess = square;
+
+      highlightTimer?.cancel();
+      highlightTimer = Timer(const Duration(milliseconds: 200), () {
+        setState(() {
+          highlightLastGuess = null;
+        });
+      });
+    });
+  }
+}
+
+class _TimeBar extends StatelessWidget {
+  const _TimeBar({
+    required this.maxWidth,
+    required this.timeFractionElapsed,
+    required this.color,
+  });
+
+  final double maxWidth;
+  final double? timeFractionElapsed;
+  final Color color;
+
+  @override
+  Widget build(BuildContext context) {
+    return Align(
+      alignment: Alignment.center,
+      child: SizedBox(
+        width: maxWidth * (timeFractionElapsed ?? 0.0),
+        height: 15.0,
+        child: ColoredBox(
+          color: color,
+        ),
+      ),
+    );
+  }
+}
+
+class _CoordinateTrainingMenu extends ConsumerWidget {
+  const _CoordinateTrainingMenu();
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    final trainingPrefs = ref.watch(coordinateTrainingPreferencesProvider);
+    final trainingPrefsNotifier =
```

Comment:
For the sake of consistency I'd convert that one to ref.read in callbacks as well.